miniapp/USERGUIDE.md

```diff
@@ -622,6 +622,43 @@ mini app scheme and should close external webview.
 
 Using `#ExternalResultHandler.emitResult(String)` to transmit the url string to mini app view.
 
+### File choosing
+**API Docs:** [MiniAppFileChooser](api/com.rakuten.tech.mobile.miniapp.file/-mini-app-file-chooser/)
+
+The mini app is able to choose the file which is requested using HTML forms with 'file' input type whenever users press a "Select file" button.
+
+- At first, HostApp should implement `MiniAppFileChooser` in the `Activity`.
+
+```kotlin
+val fileChoosingReqCode = REQUEST_CODE // define a request code in HostApp
+val miniAppFileChooser = MiniAppFileChooser(requestCode = fileChoosingReqCode)
+```
+
+- HostApp needs to pass `MiniAppFileChooser` through `MiniApp.create(appId: String, miniAppMessageBridge: MiniAppMessageBridge, miniAppFileChooser: MiniAppFileChooser)`.
+Then, HostApp activity can receive the files at `onActivityResult` as following:
+
+```kotlin
+override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    super.onActivityResult(requestCode, resultCode, data)
+
+    if (requestCode == fileChoosingReqCode && resultCode == Activity.RESULT_OK) {
+        data?.let { intent ->
+            val result: Uri? = intent.data
+            miniAppFileChooser.onReceivedFiles(arrayOf(result!!))
+        }
+    }
+}
+```
+
+Also, HostApp can pass the necessary parameters using `MiniAppFileChooser.onShowFileChooser` to choose a file as following:
+
+```kotlin
+val filePathCallback: ValueCallback<Array<Uri>>? = VALUE_HERE
+val fileChooserParams: WebChromeClient.FileChooserParams? = VALUE_HERE
+
+miniAppFileChooser.onShowFileChooser(filePathCallback, fileChooserParams, context)
+```
```

Comment:
Is this something the Host App would really want to do? 

I think a better example here would be how they can override the `onShowFileChooser` functionality.

Reply:
Yeah, I am thinking for an example how they can override it, currently, I have removed the guideline about it.